Source/Core/Core/Debugger/BranchWatch.h

```diff
@@ -0,0 +1,286 @@
+  }
+
+  static void HitVF_fk(BranchWatch* branch_watch, u64 fake_key, u32 inst)
+  {
+    branch_watch->m_collection_vf[{Common::BitCast<FakeBranchWatchCollectionKey>(fake_key), inst}]
+        .total_hits += 1;
+  }
+
+  static void HitPF_fk(BranchWatch* branch_watch, u64 fake_key, u32 inst)
+  {
+    branch_watch->m_collection_pf[{Common::BitCast<FakeBranchWatchCollectionKey>(fake_key), inst}]
+        .total_hits += 1;
+  }
+
+  static void HitVT_fk_n(BranchWatch* branch_watch, u64 fake_key, u32 inst, u32 n)
+  {
+    branch_watch->m_collection_vt[{Common::BitCast<FakeBranchWatchCollectionKey>(fake_key), inst}]
+        .total_hits += n;
+  }
+
+  static void HitPT_fk_n(BranchWatch* branch_watch, u64 fake_key, u32 inst, u32 n)
+  {
+    branch_watch->m_collection_pt[{Common::BitCast<FakeBranchWatchCollectionKey>(fake_key), inst}]
+        .total_hits += n;
+  }
+
+  // HitVF_fk_n and HitPF_fk_n are never used, so they are omitted here.
+
+  static void HitVT(BranchWatch* branch_watch, u32 origin, u32 destination, u32 inst)
+  {
+    HitVT_fk(branch_watch, FK(origin, destination), inst);
+  }
+
+  static void HitPT(BranchWatch* branch_watch, u32 origin, u32 destination, u32 inst)
+  {
+    HitPT_fk(branch_watch, FK(origin, destination), inst);
+  }
+
+  static void HitVF(BranchWatch* branch_watch, u32 origin, u32 destination, u32 inst)
+  {
+    HitVF_fk(branch_watch, FK(origin, destination), inst);
+  }
+
+  static void HitPF(BranchWatch* branch_watch, u32 origin, u32 destination, u32 inst)
+  {
+    HitPF_fk(branch_watch, FK(origin, destination), inst);
+  }
+
+  void HitT(u32 origin, u32 destination, UGeckoInstruction inst, bool translate)
+  {
+    if (translate)
+      HitVT(this, origin, destination, inst.hex);
+    else
+      HitPT(this, origin, destination, inst.hex);
+  }
+
+  void HitF(u32 origin, u32 destination, UGeckoInstruction inst, bool translate)
+  {
+    if (translate)
+      HitVF(this, origin, destination, inst.hex);
+    else
+      HitPF(this, origin, destination, inst.hex);
+  }
+
+  // The JIT needs this value, but doesn't need to be a full-on friend.
+  static constexpr int GetOffsetOfRecordingActive()
+  {
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Winvalid-offsetof"
+#endif
+    return offsetof(BranchWatch, m_recording_active);
+#ifdef __GNUC__
+#pragma GCC diagnostic pop
+#endif
+  }
+
+private:
+  static Collection BranchWatch::*FindCollectionV(bool condition)
+  {
+    if (condition)
+      return &BranchWatch::m_collection_vt;
+    return &BranchWatch::m_collection_vf;
+  }
+
+  static Collection BranchWatch::*FindCollectionP(bool condition)
+  {
+    if (condition)
+      return &BranchWatch::m_collection_pt;
+    return &BranchWatch::m_collection_pf;
+  }
+
+  static Collection BranchWatch::*FindCollection(bool is_virtual, bool condition)
+  {
+    if (is_virtual)
+      return FindCollectionV(condition);
+    return FindCollectionP(condition);
+  }
```

Comment:
Is there a reason these are static? It looks to me as if these are used in exactly one place, in a member function that already has the this pointer. So these could just return a `Collection&` and be much more standard syntax-wise.